src/types.ts

```diff
@@ -5,37 +5,94 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+export interface PluginOptions {

```

Comment:
I copied this pattern from another plugin I saw. Its not awesome with the duplication but it makes it very neat since it can validate partial configurations and then still be sure in the code that the properties exists.